functions/sg/main.py

```diff
@@ -0,0 +1,116 @@
+    scopes = ','.join([
+        'https://www.googleapis.com/auth/cloud-platform',
+        'https://www.googleapis.com/auth/userinfo.email',
+        'https://www.googleapis.com/auth/userinfo.profile'
+    ])
+    metadata_url = (
+        'http://metadata.google.internal/computeMetadata/v1/'
+        f'instance/service-accounts/default/token?scopes={scopes}'
+    )
+    metadata_headers = {'Metadata-Flavor': 'Google'}
+    r = requests.get(metadata_url, headers=metadata_headers)
+    r.raise_for_status()
+    access_token = r.json()['access_token']
+    headers = {
+        'Authorization': 'Bearer {}'.format(access_token)
+    }
+    return headers
+
+
+def filter_for_input_paths(pattern, event_file):
+    if pattern is None or re.search(pattern, event_file) is not None:
+        return event_file
+    else:
+        return None
+
+
+def make_inputs(input_file):
+    if input_file.endswith('.bam'):
+        return {'ubam': input_file}
+    else:
+        print(f'Ignoring non-input file within input path: {input_file}')
+        return None
+
+
+def make_payload(inputs):
+    return {
+        'cromwell': CROMWELL_URL,
+        'output': OUTPUT_BUCKET,
+        'pipeline': 'GDCWholeGenomeSomaticSingleSample',
+        'project': WORKLOAD_PROJECT,
+        'items': [
+            {
+                'inputs': inputs
+            }
+        ]
+    }
+
+
+def describe_workload(workload):
+    workflows = [w["uuid"] for w in workload["workflows"]]
+    print(f'Started workload {workload["uuid"]} with workflows {workflows}')
+    return workflows
+
+
+def post_payload(headers, payload):
+    try:
+        response = requests.post(
+            url=f'{WFL_URL}/api/v1/exec',
+            headers=headers,
+            json=payload
+        )
+        response.raise_for_status()
+        workload = response.json()
+        return describe_workload(workload)
+    except Exception as e:
+        print(f'The failed payload: {payload}')
+        try:
+            # noinspection PyUnboundLocalVariable
+            print(f'The failed response: {response.text}')
+        except UnboundLocalError:
+            print('No response text available')
+        raise e
+
+
+def submit_sg_workload(event, _):
+    """Background Cloud Function to be triggered by Cloud Storage.
+
+    Args:
+        event (dict):  The dictionary with data specific to this type of event.
+                       The `data` field contains a description of the event in
+                       the Cloud Storage `object` format described here:
+                       https://cloud.google.com/storage/docs/json_api/v1/objects#resource
+        _ (google.cloud.functions.Context): Metadata of triggering event.
+    """
+    headers = get_auth_headers()
+
+    input_file = filter_for_input_paths(
+        PATH_PATTERN,
+        f"gs://{event['bucket']}/{event['name']}"
+    )
+    if input_file is None:
+        return
+
+    inputs = make_inputs(input_file)
+    if inputs is None:
+        return
```

Comment:
I think it would be clearer if the logic of "only do something if this input file is a BAM" here rather than buried in an internal procedure. Perhaps just move the check 
```python
   if input_file is not None and input_file.endswith('.bam'):
       # post the payload
```

Reply:
Good idea, done